proposals/low-level-struct-improvements.md

```diff
@@ -824,19 +817,69 @@ ref struct S<T> {
+
+Examples:
+
+```c#
+Span<T> CreateSpan<T>(escapes ref T value) => new Span<T>(value)
+
+escapes struct S 
+{
+    int field;
+
+    ref int Prop => ref field;
+}
+```
+
 ### Take the breaking change
-Consider for a minute a design where the compat problem was approached from the other direction. Effectively make it such that every method was implicitly `[RefFieldsEscape]` and then have an attribute that restores the span safety rules in place today. Say `[RefFieldDoesNotEscape]`. 
+<a name="breaking"></a>
+This design is predicated on the idea that the language must remain fully compatible with the existing span safety rules. This results in a sub-optimal design for the new features to support existing APIs that are likely small in number and in several cases simply designed incorrectly. Effectively this design puts more burden on correct APIs to support legacy flaws in older ones.
+
+Consider for a minute where the design was approached from the other direction: APIs with `ref` fields have the ideal state and burden is placed on developers with incorrect APIs to fix them up when compiling for C# 11. This compat break would take primarily two forms:
+
+* A `ref` parameter on method returning a `ref struct` would be considered `[RefFieldsEscape]`.
+* A `out` parameter would be considered `ref-safe-to-escape` within the current method.
+
+This change would impact methods that have the following properties:
+
+- Have a `Span<T>` or `ref struct`
+    - Where the `ref struct` is a return type, `ref` or `out` parameter
+    - Has an additional `in` or `ref` parameter
+
+To understand the impact it's helpful to break APIs into categories:
+
+1. Want consumers to account for `ref` being captured as a `ref` field. Prime example is the `Span(ref T value)` constructors 
+2. Do not want consumers to account for `ref` being captured as a `ref` field. These though break into two catogries
+    1. Unsafe APIs. These are APIS inside the `Unsafe` and `MemoryMarshal` types, of which `MemoryMarshal.CreateSpan` is the most prominent. These APIs do capture the `ref` unsafely but they are also known to be unsafe APIs. 
+    2. Safe APIs. These are APIs which take `ref` parameters for efficiency but it is not actually captured anywhere. Examples are small but one is `AsnDecoder.ReadEnumeratedBytes`
+
+This change primarily benefits (1) above. These are expected to make up the majority of APIs that take a `ref` and return a `ref struct` going forward. The changes negatively impact (2.1) and (2.2) as it breaks the existing calling semantics because the lifetime rules change. 
+
+The APIs in category (2.1) though are largely authored by Microsoft or by developers who stand the most to benefit from `ref` fields (the Tanner's of the world). It is reasonable to assume this class of developers would be ammenable to a compatability tax on upgrade to C# 11 in the form of a few annotations to retain the existing semantics if `ref` fields were provided in return.
```

Comment:
>(the Tanner's of the world) [](http://example.com/codeflow?start=133&length=27)

Consider omitting.

Reply:
This made me giggle. 10/10

Notably, in my own case I am amenable to the cost here and it should be possible to configure tooling (such as ClangSharp) to do this "automagically". In the case of my own usages of things like `MemoryMarshal`, they were mainly a workaround for the language missing the  `RefThisEscapes` and related functionality. The intent is to switch to those features once available and make this "safe" instead.

I expect the BCL couldn't/wouldn't update their own APIs and they would remain unsafe. There would just be less reason for users to need them.

One particular area of interest is in how it impacts users multi-targeting the last LTS (`.net 6`) and the latest (`.net 7`) and the experience users would see on downstream compilers.